common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -74,12 +78,31 @@
     private static final Object sLock = new Object();
     private static InteractiveTokenCommand sCommand = null;
     private static final CommandResultCache sCommandResultCache = new CommandResultCache();
+
+    private static final Object mapAccessLock = new Object();
+    @GuardedBy("mapAccessLock")
     // Suppressing rawtype warnings due to the generic type BaseCommand
     @SuppressWarnings(WarningType.rawtype_warning)
-    private static final ConcurrentMap<BaseCommand, ResultFuture<CommandResult>> sExecutingCommandMap = new ConcurrentHashMap<>();
+    private static ConcurrentMap<BaseCommand, ResultFuture<CommandResult>> sExecutingCommandMap = new ConcurrentHashMap<>();
+
+    /**
+     * MUST be executed under the mapAccessLock;
+     * @param command
+     */
+    // Suppressing rawtype warnings due to the generic type BaseCommand
+    @SuppressWarnings(WarningType.rawtype_warning)
+    private static void cleanMap(BaseCommand command) {
```

Comment:
Can we javadoc why this function is required? Otherwise not apparent to those who didn't debug this 😄

Reply:
6d73792efa129d69a0aaa08c94322f1dd09e047f